frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionActions/QuestionActions.tsx

```diff
@@ -94,7 +92,7 @@ export const QuestionActions = ({
   const canWrite = question.canWrite();
   const isSaved = question.isSaved();
   const database = question.database();
-  const canAppend = canUpload && canWrite && !!question._card.based_on_upload;
+  const canAppend = canWrite && !!question._card.based_on_upload;
```

Comment:
This is the key change. We don't need to check that the question's database has `can_upload = true`. Using based_on_upload` is sufficient to determine whether the user can upload to the table underlying the card.

Here's the docstring for what based_on_upload means:
https://github.com/metabase/metabase/blob/eef0cecb1eaaba5592df70359af09bb3f68903da/src/metabase/upload.clj#L782-L790

We could go a step further than this and do the `canWrite` check on the BE too, so the logic is not split between frontend and backend. Then we could rename `based_on_upload` to `can_append`.